### ISSUE TITLE
haxe hide.hxml fails

### ISSUE THREAD

**Issue by kgish** [HeapsIO/hide#227]
When I follow th readme instructions and try and run the command `haxe hide.hxml` I get the following errors:

```
hide/view/l3d/Gizmo.hx:408: characters 32-41 : hxd.Window has no field mouseMode (Suggestion: mouseLock)
hide/view/l3d/Gizmo.hx:408: characters 48-63 : Unknown identifier : AbsoluteUnbound
hide/view/l3d/Gizmo.hx:408: characters 72-80 : Unknown identifier : Absolute
hrt/prefab/l3d/MeshSpray.hx:217: characters 10-29 : Null<h3d.scene.MeshBatch> has no field alwaysSyncAnimation
hrt/prefab/l3d/MeshSpray.hx:246: characters 10-29 : h3d.scene.Object has no field alwaysSyncAnimation
hrt/prefab/l3d/Spray.hx:85: characters 10-29 : h3d.scene.Object has no field alwaysSyncAnimation
hrt/prefab/fx/FX.hx:112: characters 37-56 : Unknown identifier : alwaysSyncAnimation
hrt/prefab/fx/Emitter.hx:552: characters 10-20 : h3d.scene.MeshBatch has no field calcBounds
hrt/prefab/Light.hx:146: characters 63-74 : h3d.scene.Object should be Null<h3d.Vector>
hrt/prefab/Light.hx:146: characters 63-74 : ... For optional function argument 'dir'
hrt/prefab/Light.hx:176: characters 31-56 : Type not found : h3d.pass.CascadeShadowMap
hide/view/CameraController.hx:42: characters 33-42 : hxd.Window has no field mouseMode (Suggestion: mouseLock)
hide/view/CameraController.hx:42: characters 45-60 : Unknown identifier : AbsoluteUnbound
hide/view/CameraController.hx:49: characters 34-43 : hxd.Window has no field mouseMode (Suggestion: mouseLock)
hide/view/CameraController.hx:49: characters 46-54 : Unknown identifier : Absolute
hide/view/CameraController.hx:88: characters 33-42 : hxd.Window has no field mouseMode (Suggestion: mouseLock)
hide/view/CameraController.hx:88: characters 45-53 : Unknown identifier : Absolute
hide/view/l3d/Gizmo.hx:406: characters 63-72 : hxd.Window has no field mouseMode (Suggestion: mouseLock)
hide/view/l3d/Gizmo.hx:406: characters 76-84 : Unknown identifier : Absolute
```

**Comment by Speedphoenix** [comment]
are you up to date on your heaps version?
https://github.com/HeapsIO/heaps

**Comment by ernesernesto** [comment]
Also fails using latest heaps from git (1.10.0), will show this errors

`hrt/prefab/fx/BaseFX.hx:194: characters 29-44 : Null<h3d.scene.MeshBatch> has no field instancedParams
hrt/prefab/fx/BaseFX.hx:196: characters 24-54 : Module hxsl.Cache does not define type BatchInstanceParams
hrt/prefab/fx/BaseFX.hx:197: characters 12-27 : Null<h3d.scene.MeshBatch> has no field instancedParams
hrt/prefab/l3d/Trails.hx:739: characters 16-28 : h3d.Buffer has no field uploadFloats
hrt/prefab/fx/Emitter.hx:1295: characters 20-32 : Field addBoundsRec is declared 'override' but doesn't override any field (Suggestion: getBoundsRec)
hrt/prefab/l3d/HeightMap.hx:1166: characters 4-36 : Type not found : hxd.BufferFormat
hrt/prefab/l3d/HeightMap.hx:1168: characters 4-33 : Type not found : hxd.BufferFormat
hrt/prefab/l3d/HeightMap.hx:1170: characters 4-29 : Type not found : hxd.BufferFormat
hrt/prefab/l3d/HeightMap.hx:1172: characters 4-26 : Type not found : hxd.BufferFormat
hrt/prefab/l3d/HeightMap.hx:1165: characters 7-13 : Variables of type Void are not allowed
hrt/prefab/l3d/HeightMap.hx:1173: characters 70-76 : Void has no field stride
hrt/prefab/l3d/HeightMap.hx:1173: characters 33-76 : Float should be Int
hrt/prefab/l3d/HeightMap.hx:1173: characters 33-76 : ... For optional function argument 'length'
hrt/prefab/l3d/HeightMap.hx:1190: characters 37-43 : Void should be Int
hrt/prefab/l3d/HeightMap.hx:1190: characters 37-43 : ... For function argument 'stride'
hrt/prefab/l3d/Trails.hx:433: characters 13-45 : Type not found : hxd.BufferFormat
hrt/prefab/l3d/SplineMesh.hx:107: characters 26-42 : Type not found : hxd.BufferFormat`

**Comment by andresmakesgames** [comment]
I'm also experiencing this same issue.

**Comment by saklis** [comment]
I'm getting different error while building:
`haxe/lib/heaps/git/hxd/res/Image.hx:318: characters 22-29 : Class<format.png.Tools> has no field extract (Suggestion: extract32)`

**Comment by kLabz** [comment]
You need git version of `format` lib (https://github.com/HaxeFoundation/format)

**Comment by saklis** [comment]
> You need git version of `format` lib (https://github.com/HaxeFoundation/format)

Works!
Missed it because it wasn't mentioned in readme.

Thank you!

**Comment by melMass** [comment]
Hi, this is still an issue when using:
```sh
haxelib git heaps https://github.com/HeapsIO/heaps.git
```

As haxelib installs the non git version of format (3.6)

Running 
```sh
haxelib git format https://github.com/HaxeFoundation/format
haxelib update
```

Does fix it.

**Comment by rbobka** [comment]
I attempted to build HIDE using the instructions.  I was given the error message:

> hrt/impl/Gradient.hx:5: characters 8-19 : Type not found : h3d.Vector4

Is there a library that I'm missing?  I thought that h3d had a Vector but I don't know where to find a Vector4.

**Comment by ncannasse** [comment]
You need latest heaps master to compile hide.

**Comment by rbobka** [comment]
Thank you Nicolas, after I posted this I did a deep dive and saw that you had made some changes to Heaps in the past month including vectors.  I guess my real problem is that haxelib update does not seem to be updating my libraries (even though it says it is).

**Comment by ncannasse** [comment]
I haven't made a haxelib release recently. Will do later this year or early
next year.

**Comment by yz-xlame** [comment]
1. remove all package
```
haxelib remove heaps
haxelib remove castle
haxelib remove hxbit
haxelib remove hscript
haxelib remove hxnodejs
haxelib remove domkit
haxelib remove hx3compat
```
2. install all package
```
haxelib git heaps https://github.com/HeapsIO/heaps
haxelib git castle https://github.com/ncannasse/castle
haxelib git hxbit https://github.com/ncannasse/hxbit
haxelib git hscript https://github.com/HaxeFoundation/hscript
haxelib git hxnodejs https://github.com/HaxeFoundation/hxnodejs
haxelib git domkit https://github.com/HeapsIO/domkit
haxelib git hx3compat https://github.com/HaxeFoundation/hx3compat
```
work for me

**Comment by kLabz** [comment]
Yeah works again since `format` 3.7.0 has been released since then

**Comment by Cjkjvfnby** [comment]
To address `Type not found : hlprof.ProfileGen` I've updated installation instructions here  https://github.com/HeapsIO/hide/pull/277


To make things a bit easier for users, it's possible to move some dependencies to the https://github.com/HeapsIO/hide/blob/master/common.hxml

It will look like this:
```
-lib format:git:https://github.com/HaxeFoundation/format
-lib heaps:git:https://github.com/HeapsIO/heaps
-lib castle:git:https://github.com/ncannasse/castle
-lib hxbit:git:https://github.com/ncannasse/hxbit
-lib hscript:git:https://github.com/HaxeFoundation/hscript
-lib hxnodejs:git:https://github.com/HaxeFoundation/hxnodejs
-lib domkit:git:https://github.com/HeapsIO/domkit
-lib hx3compat:git:https://github.com/HaxeFoundation/hx3compat
```

And install them with the `haxelib install hide.hxml`  This will work for almost all dependencies, except: `haxelib git hashlink https://github.com/HaxeFoundation/hashlink master other/haxelib/`, because subdirs are not supported when installing in such a way.

To ensure that all libs are in the file, the nightly build should use `yes | haxelib install hide.hxml` instead of duplicating all dependencies  https://github.com/HeapsIO/hide/blob/master/.github/workflows/main.yml#L22  (still need to install hashlink in an old way)

This approach will keep all dependencies in a single place.
